File: RepoTransBench/target_projects/Java/C#/tamir7_Contacts/tests/original/EventTest.cs
using Xunit;
using Contacts;

namespace Contacts.OriginalTests
{
    public class EventTest
    {
        [Fact]
        public void TestConstructorsAndGetters_Type()
        {
            var e = new Event("2023-01-01", Event.Type.BIRTHDAY);
            Assert.Equal("2023-01-01", e.StartDate);
            Assert.Equal(Event.Type.BIRTHDAY, e.TypeValue);
            Assert.Null(e.Label);
        }

        [Fact]
        public void TestConstructorsAndGetters_Label()
        {
            var e = new Event("2023-01-01", "Anniversary");
            Assert.Equal("2023-01-01", e.StartDate);
            Assert.Equal(Event.Type.CUSTOM, e.TypeValue);
            Assert.Equal("Anniversary", e.Label);
        }

        [Fact]
        public void TestEqualsAndHashCode()
        {
            var e1 = new Event("2020-10-10", Event.Type.BIRTHDAY);
            var e2 = new Event("2020-10-10", Event.Type.BIRTHDAY);
            var e3 = new Event("2020-10-10", "CustomEvt");
            Assert.Equal(e1, e2);
            Assert.NotEqual(e1, e3);
            Assert.Equal(e1.GetHashCode(), e2.GetHashCode());
        }

        [Fact]
        public void TestTypeFromValue()
        {
            Assert.Equal(Event.Type.CUSTOM, Event.fromValue(0));
            Assert.Equal(Event.Type.ANNIVERSARY, Event.fromValue(1));
            Assert.Equal(Event.Type.OTHER, Event.fromValue(2));
            Assert.Equal(Event.Type.BIRTHDAY, Event.fromValue(3));
            Assert.Equal(Event.Type.UNKNOWN, Event.fromValue(99));
        }

        [Fact]
        public void TestNotEqualConditions()
        {
            var e1 = new Event("d", Event.Type.BIRTHDAY);
            Assert.False(e1.Equals(null));
            Assert.False(e1.Equals("notAnEvent"));
            var e2 = new Event("other", Event.Type.BIRTHDAY);
            Assert.NotEqual(e1, e2);
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/tamir7_Contacts/tests/original/event_test.go
package original

import (
	"testing"
	"github.com/tamir7/contacts"
)

func TestEvent_ConstructorsAndGetters_Type(t *testing.T) {
	e := contacts.NewEventWithType("2023-01-01", contacts.EventTypeBIRTHDAY)
	if e.StartDate != "2023-01-01" {
		t.Errorf("expected startDate '2023-01-01', got %v", e.StartDate)
	}
	if e.Type != contacts.EventTypeBIRTHDAY {
		t.Errorf("expected type BIRTHDAY, got %v", e.Type)
	}
	if e.Label != "" {
		t.Errorf("expected label empty, got %v", e.Label)
	}
}

func TestEvent_ConstructorsAndGetters_Label(t *testing.T) {
	e := contacts.NewEventWithLabel("2023-01-01", "Anniversary")
	if e.StartDate != "2023-01-01" {
		t.Errorf("expected '2023-01-01', got %v", e.StartDate)
	}
	if e.Type != contacts.EventTypeCUSTOM {
		t.Errorf("expected type CUSTOM, got %v", e.Type)
	}
	if e.Label != "Anniversary" {
		t.Errorf("expected label Anniversary, got %v", e.Label)
	}
}

func TestEvent_EqualsAndHashCode(t *testing.T) {
	e1 := contacts.NewEventWithType("2020-10-10", contacts.EventTypeBIRTHDAY)
	e2 := contacts.NewEventWithType("2020-10-10", contacts.EventTypeBIRTHDAY)
	e3 := contacts.NewEventWithLabel("2020-10-10", "CustomEvt")
	if !e1.Equal(e2) {
		t.Errorf("expected e1 == e2")
	}
	if e1.Equal(e3) {
		t.Errorf("expected e1 != e3")
	}
	if e1.HashCode() != e2.HashCode() {
		t.Errorf("hashcodes should match")
	}
}

func TestEvent_TypeFromValue(t *testing.T) {
	if contacts.EventTypeFromValue(0) != contacts.EventTypeCUSTOM {
		t.Errorf("expected 0->CUSTOM")
	}
	if contacts.EventTypeFromValue(1) != contacts.EventTypeANNIVERSARY {
		t.Errorf("expected 1->ANNIVERSARY")
	}
	if contacts.EventTypeFromValue(2) != contacts.EventTypeOTHER {
		t.Errorf("expected 2->OTHER")
	}
	if contacts.EventTypeFromValue(3) != contacts.EventTypeBIRTHDAY {
		t.Errorf("expected 3->BIRTHDAY")
	}
	if contacts.EventTypeFromValue(99) != contacts.EventTypeUNKNOWN {
		t.Errorf("expected 99->UNKNOWN")
	}
}

func TestEvent_NotEqualConditions(t *testing.T) {
	e1 := contacts.NewEventWithType("d", contacts.EventTypeBIRTHDAY)
	if e1.Equal(nil) {
		t.Errorf("should not equal nil")
	}
	if e1.Equal("notAnEvent") {
		t.Errorf("should not equal a string")
	}
	e2 := contacts.NewEventWithType("other", contacts.EventTypeBIRTHDAY)
	if e1.Equal(e2) {
		t.Errorf("should not equal another event with different StartDate")
	}
}